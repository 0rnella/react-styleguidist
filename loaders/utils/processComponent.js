'use strict';

const path = require('path');
const getNameFallback = require('./getNameFallback');
const getExamples = require('./getExamples');
const requireIt = require('./requireIt');
const toCode = require('./toCode');

/**
 * Return JS code as a string for a component with all required for style guide information.
 *
 * @param {string} filepath
 * @param {object} config
 * @returns {string}
 */
module.exports = function processComponent(filepath, config) {
	const nameFallback = getNameFallback(filepath);
	const examplesFile = config.getExampleFilename(filepath);
	const componentPath = path.relative(config.configDir, filepath);
	return toCode({
		filepath: JSON.stringify(componentPath),
		nameFallback: JSON.stringify(nameFallback),
		pathLine: JSON.stringify(config.getComponentPathLine(componentPath)),
		module: requireIt(filepath),
		props: requireIt('!!../loaders/props-loader!' + filepath),
		examples: getExamples(examplesFile, nameFallback, config.defaultExample),
	});
};
