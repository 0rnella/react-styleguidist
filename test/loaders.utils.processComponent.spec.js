import vm from 'vm';
import path from 'path';
import processComponent from '../loaders/utils/processComponent';

const config = {
	configDir: __dirname,
	getExampleFilename: componentpath => path.join(path.dirname(componentpath), 'Readme.md'),
	getComponentPathLine: componentpath => componentpath,
};

it('processComponent() should return valid JS for section with content', () => {
	const result = processComponent('pizza.js', config);

	expect(new vm.Script(`a=${result}`)).not.toThrowError(SyntaxError);
	expect(result).toMatchSnapshot();
});
