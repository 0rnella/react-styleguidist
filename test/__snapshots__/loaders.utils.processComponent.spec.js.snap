exports[`test processComponent() should return valid JS for section with content 1`] = `
"{\'filepath\': \"../pizza.js\",
\'nameFallback\': \"pizza\",
\'pathLine\': \"../pizza.js\",
\'module\': require(\"pizza.js\"),
\'props\': require(\"!!props-loader!pizza.js\"),
\'examples\': require(\"!!examples-loader!Readme.md\")}"
`;
